Retroarch reports "Core Installed" when installation fails.

# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

My libretro directory is not writeable.  Cores obviously fail to install, but the notification says "Core Installed"

### Expected behavior

The GUI should produce an error message.  The log contains an appropriate error message.
[ERROR] [core updater] Download of '/usr/lib/x86_64-linux-gnu/libretro/fceumm_libretro.so.zip' failed: Write failed.

### Actual behavior

Retroarch reports "Core Installed", when it failed to even write the archive, let alone extract and install it. 

### Steps to reproduce the bug
Install RetroArch
Remove write permissions from your libretro directory
Install cores with the Core Downloader

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit]

### Environment information

- OS: Debian Linux

--- a/tasks/task_core_updater.c
+++ b/tasks/task_core_updater.c
@@ -77,6 +77,7 @@ enum core_updater_download_status
    CORE_UPDATER_DOWNLOAD_START_TRANSFER,
    CORE_UPDATER_DOWNLOAD_WAIT_TRANSFER,
    CORE_UPDATER_DOWNLOAD_WAIT_DECOMPRESS,
+   CORE_UPDATER_DOWNLOAD_ERROR,
    CORE_UPDATER_DOWNLOAD_END
 };
 
@@ -633,9 +634,11 @@ void cb_http_task_core_updater_download(
 finish:
    /* Log any error messages */
    if (!string_is_empty(err))
+   {
       RARCH_ERR("[core updater] Download of '%s' failed: %s\n",
             (transf ? transf->path: "unknown"), err);
-
+      download_handle->status = CORE_UPDATER_DOWNLOAD_ERROR;
+   }
    if (transf)
       free(transf);
 
@@ -921,6 +924,21 @@ static void task_core_updater_download_handler(retro_task_t *task)
              * callback to trigger */
             if (download_handle->decompress_task_complete)
                download_handle->status = CORE_UPDATER_DOWNLOAD_END;
+         }
+         break;
+      case CORE_UPDATER_DOWNLOAD_ERROR:
+         {
+            char task_title[PATH_MAX_LENGTH];
+
+            /* Set final task title */
+            task_free_title(task);
+
+            strlcpy(task_title, msg_hash_to_str(MSG_CORE_INSTALL_FAILED), sizeof(task_title));
+            strlcat(task_title, download_handle->display_name, sizeof(task_title));
+
+            task_set_title(task, strdup(task_title));
+            task_set_progress(task, 100);
+            goto task_finished;
          }
          break;
       case CORE_UPDATER_DOWNLOAD_END:

